Origin: http://websvn.kde.org/?view=revision&revision=983650
Description: fix playback of files with hash symbols (#) in their name
Bug: https://bugs.kde.org/show_bug.cgi?id=172242
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=546271

--- a/xine/mediaobject.cpp
+++ b/xine/mediaobject.cpp
@@ -321,7 +321,7 @@
 {
     for (int i = 0; i < mrl.size(); ++i) {
         const unsigned char c = static_cast<unsigned char>(mrl.at(i));
-        if (c & 0x80 || c == '\\' || c < 32 || c == '%') {
+        if (c & 0x80 || c == '\\' || c < 32 || c == '%' || c == '#') {
             char enc[4];
             qsnprintf(enc, 4, "%%%02X", c);
             mrl = mrl.left(i) + QByteArray(enc, 3) + mrl.mid(i + 1);
